test/Makefile

```diff
@@ -110,21 +110,40 @@ DRUNTIME_PATH=..\..\druntime
 PHOBOS_PATH=..\..\phobos
 export DFLAGS=-I$(DRUNTIME_PATH)\import -I$(PHOBOS_PATH)
 export LIB=$(PHOBOS_PATH)
+
+# auto-tester might run the testsuite with a different $(MODEL) than DMD
+# has been compiled with. Hence we manually check which binary exists.
+# For windows the $(OS) during build is: `windows`
+ifeq (,$(wildcard ../generated/windows/$(BUILD)/64/dmd$(EXE)))
+DMD_MODEL=32
+else
+DMD_MODEL=64
+endif
+export DMD=../generated/windows/$(BUILD)/$(DMD_MODEL)/dmd$(EXE)
+
 else
 export ARGS=-inline -release -g -O -fPIC
-export DMD=../src/dmd
 export EXE=
 export OBJ=.o
 export DSEP=/
 export SEP=/
 
+# auto-tester might run the testsuite with a different $(MODEL) than DMD
+# has been compiled with. Hence we manually check which binary exists.
+ifeq (,$(wildcard ../generated/$(OS)/$(BUILD)/64/dmd))
+DMD_MODEL=32
+else
+DMD_MODEL=64
+endif
+export DMD=../generated/$(OS)/$(BUILD)/$(DMD_MODEL)/dmd
+
 DRUNTIME_PATH=../../druntime
 PHOBOS_PATH=../../phobos
 # link against shared libraries (defaults to true on supported platforms, can be overridden w/ make SHARED=0)
 SHARED=$(if $(findstring $(OS),linux freebsd),1,)
-DFLAGS=-I$(DRUNTIME_PATH)/import -I$(PHOBOS_PATH) -L-L$(PHOBOS_PATH)/generated/$(OS)/release/$(MODEL)
+DFLAGS=-I$(DRUNTIME_PATH)/import -I$(PHOBOS_PATH) -L-L$(PHOBOS_PATH)/generated/$(OS)/$(BUILD)/$(MODEL)
```

Comment:
Druntime and Phobos are built with the opposite MODEL on auto-tester
For example for 32_64: builds 32-bit DMD, but 64-but DRuntime + Phobos